File: ruby/lib/conditionals.rb
module Conditionals
  def self.demonstrate
    puts "\n=== Conditional Statements ==="
    
    # if-elsif-else statement
    number = 15
    if number > 0
      puts "#{number} is positive"
    elsif number < 0
      puts "#{number} is negative"
    else
      puts "The number is zero"
    end
    
    # One-line if (modifier form)
    puts "#{number} is even" if number.even?
    
    # unless statement (opposite of if)
    unless number.zero?
      puts "The number is not zero"
    end
    
    # case statement (switch)
    grade = 'A'
    case grade
    when 'A'
      puts "Excellent!"
    when 'B'
      puts "Good job!"
    when 'C'
      puts "Fair"
    else
      puts "Invalid grade"
    end
    
    # Ternary operator
    is_raining = true
    weather = is_raining ? "Bring an umbrella" : "Enjoy the sun"
    puts "\nWeather advice: #{weather}"
    
    # Ruby's special operators for conditional assignment
    # ||= will only assign if the variable is nil or false
    name = nil
    name ||= "Guest"
    puts "\nDefault name: #{name}"
    
    # Using and/or for control flow
    puts "\nUsing and/or:"
    puts "Success" and return if true
    puts "Failure" or return if false
    
    # Statement modifiers
    puts "It's sunny!" unless is_raining
    
    # Using begin/end with rescue (exception handling)
    begin
      result = 10 / 0
    rescue ZeroDivisionError
      puts "\nCannot divide by zero!"
    end
  end
end

File: fortran/src/conditionals.f90
module conditionals_mod
    implicit none

contains
    subroutine demonstrate_conditionals()
        integer :: number, score
        character(len=1) :: grade
        logical :: is_raining
        
        print *, '=== Conditional Statements ==='
        
        ! If-Then-Else statement
        number = 15
        if (number > 0) then
            print *, number, ' is positive'
        else if (number < 0) then
            print *, number, ' is negative'
        else
            print *, 'The number is zero'
        end if
        
        ! Case statement (Select Case)
        score = 85
        select case (score)
            case (90:100)
                grade = 'A'
            case (80:89)
                grade = 'B'
            case (70:79)
                grade = 'C'
            case default
                grade = 'F'
        end select
        
        print *, 'Grade: ', grade
        
        ! Logical IF statement (single line)
        is_raining = .true.
        if (is_raining) print *, 'Bring an umbrella!'
        
        ! Nested if statements
        if (score >= 60) then
            if (score >= 90) then
                print *, 'Excellent performance!'
            else
                print *, 'Good performance'
            end if
        else
            print *, 'Need improvement'
        end if
    end subroutine demonstrate_conditionals
end module conditionals_mod